package-testing/sdk-test-runner/src/app.ts

```diff
@@ -8,35 +8,89 @@ import { BanditActionRequest } from './dto/banditActionRequest';
 import { TestResponse } from './dto/testResponse';
 import { Scenario, Scenarios } from './dto/scenario';
 import { TestCase, TestSuite, TestSuiteReport, getJunitXml } from 'junit-xml';
+import {
+  ClientSDKRelay,
+  FeatureNotSupportedError,
+  SDKConnectionFailure,
+  SDKInfo,
+  SDKRelay,
+  ServerSDKRelay,
+} from './protocol';
+import { Config, SdkType } from './config';
 
 export default class App {
-  public constructor(
-    private readonly sdkName: string,
-    private readonly sdkServer: string,
-    private readonly apiServer: string,
-    private readonly testDataPath: string,
-    private readonly scenarioFile: string,
-    private readonly junitFile: string,
-  ) {}
+  private readonly config: Config;
 
-  private printHeader(): void {
-    log(`Firing up test runner for ${this.sdkName}`);
-    log(`Posting test cases to SDK server at ${this.sdkServer}`);
-    log(`Controlling configuration server at ${this.apiServer}`);
+  public constructor(config: Config) {
+    this.config = config;
   }
 
   public async run() {
     this.printHeader();
+    if (this.config.sdkType == SdkType.SERVER) {
+      return this.runServerTest();
+    } else {
+      return this.runClientTest();
+    }
```

Comment:
I prefer ternaries, but to each their own!
```ts
return this.config.sdkType == SdkType.SERVER ? this.runServerTest() : this.runClientTest();
```

Reply:
refactored to something I think is a lot more readable and organized.